lib/puppet_x/foreman/common.rb

```diff
@@ -1,16 +1,10 @@
 module PuppetX
   module Foreman
     module Common
-      FOREMAN_HOST_PARAMS = Proc.new do
-        ensurable
-
-        newparam(:name, :namevar => true) do
-          desc 'The name of the resource.'
-        end
 
-        newparam(:hostname) do
-          desc 'The name of the host.'
-        end
+      # Parameters common to several types that use the rest_v3 api provider
+      FOREMAN_COMMON_PARAMS = Proc.new do
```

Comment:
Perhaps this could be named `REST_API_COMMON_PARAMS`?

Reply:
Done.